tools/c-parser/PackedTypes.thy

```diff
@@ -960,4 +960,69 @@ proof (simp add: packed_type_access_ti, rule ext)
   finally show "?LHS = ?RHS" .
 qed
 
+subsection {* Proof automation for packed types *}
+
+definition td_packed :: "'a field_desc typ_desc \<Rightarrow> nat \<Rightarrow> nat \<Rightarrow> bool"
+  where "td_packed t sz al \<longleftrightarrow>
+    td_fafu_idem t \<and> td_fa_hi t \<and> aggregate t \<and> size_td t = sz \<and> align_td t = al"
```

Comment:
It is possible that the size_td and align_td calculated here could be used instead of the explicit proofs.